### ISSUE TITLE
jacoco should itself build with gradle so new java versions are immediate supported

### ISSUE THREAD

**Issue by soloturn** [jacoco/jacoco#1764]
java-23 oes not work with jacoco. the gradle plugin is outdated, setting the tool.version to 0.8.13.202410300952 fails as well. to avoid this brawl with every new java version, jacoco should build itself with gradle.

**Comment by marchof** [comment]
@soloturn Please report this to the Grade plugin project. Please understand that we don't have resources to maintain another tool integration here at the JaCoCo project.

**Comment by soloturn** [comment]
@marchof which maven repo one would download the jaoco version working with java-23 ?

**Comment by Godin** [comment]
@soloturn JaCoCo version 0.8.12 from Maven Central repository

**Comment by Godin** [comment]
> java-23 oes not work with jacoco. the gradle plugin is outdated

This is a weird statement because the latest released as of today Gradle version `8.10.2` comes with the latest released JaCoCo version `0.8.12` (see https://github.com/gradle/gradle/blob/v8.10.2/platforms/jvm/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoPlugin.java#L73) and this combination works with Java 23 - here is [example.zip](https://github.com/user-attachments/files/17618698/example.zip) which explicitly requires Java 23 and for which execution of `./gradlew test jacocoTestReport` using

```
openjdk 23.0.1 2024-10-15
OpenJDK Runtime Environment Zulu23.30+13-CA (build 23.0.1+11)
OpenJDK 64-Bit Server VM Zulu23.30+13-CA (build 23.0.1+11, mixed mode, sharing)
```

works and produces expected `build/reports/jacoco/test/html/index.html`

<img width="1076" alt="Screenshot 2024-11-04 at 14 33 03" src="https://github.com/user-attachments/assets/6e816037-80e2-4218-8ff7-5600a0866a0f">

----

> setting the tool.version to 0.8.13.202410300952 fails as well

In the same example replacement of

```
repositories {
    mavenCentral()
}
```

on

```
jacoco {
    toolVersion = "0.8.13-SNAPSHOT"
}

repositories {
    maven {
        url = uri("https://oss.sonatype.org/content/repositories/snapshots")
        mavenContent {
            snapshotsOnly()
            includeGroup("org.jacoco")
        }
    }
    mavenCentral()
}
```

demonstrates how `jacoco.toolVersion` can be used for SNAPSHOT builds of not-yet-released JaCoCo version `0.8.13`

----

> to avoid this brawl with every new java version, jacoco should build itself with gradle

This statement is also weird - usually there is a lag after the availability of the first JDK Early Access build and its support in Gradle. For example https://docs.gradle.org/8.10.2/userguide/compatibility.html states

> JVM 24 and later versions are not yet supported

whereas with our current Maven-based JaCoCo build we were able to use Early Access JDK builds immediately when they appear.
